CHANGELOG.md

```diff
@@ -1,5 +1,11 @@
 # bedrock-edv-storage ChangeLog
 
+## 12.1.0 - 2022-03-xx
```

Comment:
Changing the default for TTL on delegation will be a breaking change.

Reply:
This change is additive. Increasing the TTL is a less restrictive change. Zero changes are required for systems invoking zcaps that have a TTL on their delegation less than a year. Since, the previous TTL was 90 days, that means all systems were under a year. No changes are needed.

I'd agree that this is a breaking change if the TTL was decreased.

Reply:
If one had an `edv-storage` system deployed prior to this release that used the default 90 TTL.  Do you think it would have been appropriate for one to write a test to ensure that the 90 TTL was working properly?  The test would have involved submitting a zcap with a TTL longer than 90 days and an error from the zcap validation would be expected.  Would that same test pass with this change?  I believe the answer is no, and therefore this patch has broken the test suite and therefore I believe this is a breaking change.

The idea that a change that makes something "more permissive" is somehow non breaking has come up before.  While this may be valid in some cases, we must also consider deployments that *depend* on the more restrictive, less permissive prior behavior.

Reply:
I believe I understand your perspective. If clients depend on the "more restrictive" TTL, then making it "more permissive" is breaking. Is that correct? If so, then in this specific case it should still be a non-issue. This code sets the upper bound for the TTL that clients can use. Clients that have configured their zcap delegation with an expiration less than 90 days can continue to do so.

This change on the server doesn't force clients to have more permissive zcap TTL. The "more/less permissiveness" is controlled by the client. They could drop to 80 days and that would be fine.

The test suite is a great example to think about as well. I'm on the fence. If one wrote a negative test, with a zcap that has a TTL over 90 days, then our current change was deployed... that test should fail (assuming the test case used a TTL under a year). My understanding of your perspective is if that if their test fails, we introduced a breaking change. If they did not have to update their application code, is it still a breaking change? I've always looked at breaking/non-breaking changes through the lens of the impact to application code and not test suites. The former is okay, the latter is problematic. An example is below.

If we had an API that returns `{foo: "one", bar: "two"}`. Next, a consumer made an assertion that they expect only "foo" and "bar". We make, IMO, a non-breaking change and add "baz". We now return `{foo: "one", bar: "two", baz: "three"}`. Their test suite fails, but was that a breaking change on our part?